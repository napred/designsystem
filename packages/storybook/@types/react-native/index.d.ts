// hack because react-native conflicts with @types/node and lib.dom
declare module 'react-native' {
  import { ComponentType } from 'react';

  type Style = {
    default: any;
    create(obj: any): any;
  };

  const StyleSheet: Style;
  const Image: ComponentType;
  const Text: ComponentType;
  const View: ComponentType;

  export { Image, StyleSheet, Text, View };

  export type RegisteredStyle<T> = any;
}
