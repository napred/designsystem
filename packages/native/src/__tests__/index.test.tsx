import React from 'react';
import { View } from 'react-native';
import { render } from 'react-native-testing-library';
import { createComponent, DesignSystem } from '../';

const DSView = createComponent('View', View);

it('works in react-native', () => {
  const { toJSON } = render(
    <DesignSystem>
      <DSView />,
    </DesignSystem>,
  );

  expect(toJSON()).toMatchSnapshot();
});

it('passes ref', () => {
  const ref = jest.fn();
  render(
    <DesignSystem>
      <DSView ref={ref} />,
    </DesignSystem>,
  );

  expect(ref).toHaveBeenCalledTimes(1);
});
