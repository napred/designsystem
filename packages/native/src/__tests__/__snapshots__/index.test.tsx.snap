// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works in react-native 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "fontFamily": "Palanquin",
          "fontSize": 16,
        },
      ]
    }
  />,
  ",",
]
`;
